mflowgen/test/test_module.sh

```diff
@@ -297,79 +280,140 @@ if [ "$update_cache" ]; then
+    set -x; make list | awk '$NF == "'$modname'" {print}'; set +x
+    modpfx=`make list | awk '$NF == "'$modname'" {print $2 "-"}'`
+    dirname=$modpfx$modname; # E.g. "1-Tile_PE"
+    echo "--- ...BUILD SUBGRAPH '$dirname'"
+    
+    echo "mkdir $dirname; cd $dirname"
     mkdir $dirname; cd $dirname
+    
+    echo "mflowgen run --design $garnet/mflowgen/$modname"
     mflowgen run --design $garnet/mflowgen/$modname
-    set +x
 }
-# E.g. build_module full_chip; build_module tile_array; build_module Tile_PE
-for m in ${modlist[@]}; do 
-    build_module $m;
-    final_module=$m
+
+# First mod in list is the primary module; rest are subgraphs. Right?
+# Also: in general, first mod should be "full_chip" but not currently enforced.
+# E.g. modlist=(full_chip tile_array Tile_PE)
+
+[ "$MFLOWGEN_PATH" ] || echo "WARNING MFLOWGEN_PATH var not set."
+    
+# Top level
+firstmod=${modlist[0]}
+build_module $firstmod
+
+# Subgraphs
+subgraphs=${modlist[@]:1}
+for sg in $subgraphs; do
+    build_subgraph $sg
+    echo sg=$sg
+done
+
+# Function to expand step aliases
+# E.g. 'step_alias syn' returns 'synopsys-dc-synthesis'
+function step_alias {
+    case "$1" in
+        syn)       s=synopsys-dc-synthesis ;;
+        synthesis) s=synopsys-dc-synthesis ;;
+
+        init)      s=cadence-innovus-init      ;;
+        cts)       s=cadence-innovus-cts       ;;
+        place)     s=cadence-innovus-place     ;;
+        route)     s=cadence-innovus-route     ;;
+        postroute) s=cadence-innovus-postroute ;;
+
+        gds)       s=mentor-calibre-gdsmerge ;;
+        tape)      s=mentor-calibre-gdsmerge ;;
+        merge)     s=mentor-calibre-gdsmerge ;;
+        gdsmerge)  s=mentor-calibre-gdsmerge ;;
+
+        lvs)       s=mentor-calibre-lvs ;;
+        drc)       s=mentor-calibre-drc ;;
+
+        *)         s="$1" ;;
+    esac
+
+    # Catch-all maybe? Grab the *first* hit to aviod e.g. all the "debug-" aliases
+    s=`make list |& egrep -- "$s"'$' | awk '{ print $NF; exit }'`
+    echo $s
+}
+
+echo "STEPS to take"
+for step in ${build_sequence[@]}; do
+    echo "  $step -> `step_alias $step`"
 done
 
+# TEST
+# for step in -route route rdl timing-signoff; do
+#     echo "FOOOO  '$step' -> '`step_alias $step`'"
+
 ##############################################################################
 # Copy pre-built steps from (gold) cache, if requested via '--use_cached'
 if [ "$copy_list" ]; then 
-    echo "+++ ......SETUP context from gold cache (`date +'%a %H:%M'`)"
-
-    # Build the path to the gold cache
-    gold=/sim/buildkite-agent/gold
-    for m in ${modlist[@]}; do 
-        ls $gold/*${m} >& /dev/null || echo FAIL
-        if [ "$FAIL" == "true" ]; then
-            echo "***ERROR could not find cache dir '$gold'"; exit 13; fi
-        gold=`cd $gold/*${m}; pwd`
-    done
-    [ "$DEBUG" ] && echo "  Found gold cache directory '$gold'"
+    stash_dir=/sim/buildkite-agent/stash/2020-0806-mflowgen-stash-8b4ada
```

Comment:
How is this stash being used? Who's populating it?

Reply:
It's not being used yet, this is prep for further improvements down the line. Eventually (very soon I hope) we will unstash steps instead of simply copying them from the gold directory...